cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/documentstore/InternalDocumentRevision.java

```diff
@@ -204,7 +202,7 @@ public int getGeneration() {
 
     @Override
     public ChangeNotifyingMap<String, Attachment> getAttachments() {
-        return attachments;
+        return (ChangeNotifyingMap<String, Attachment>) attachments;
```

Comment:
this cast looks ugly compared to using a more specific type on the field - what's the reason for it?